mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/mail/InMemoryMessageIdMapper.java

```diff
@@ -65,6 +68,19 @@ public InMemoryMessageIdMapper(MailboxMapper mailboxMapper, InMemoryMessageMappe
             .block();
     }
 
+    @Override
+    public Publisher<ComposedMessageIdWithMetaData> findMetadata(MessageId messageId) {
+        return mailboxMapper.list()
+            .flatMap(mailbox -> messageMapper.findInMailboxReactive(mailbox, MessageRange.all(), MessageMapper.FetchType.Full, UNLIMITED))
```

Comment:
should we have a concurrency value for this flatMap ?

Reply:
It's in memory, meaning we are not really limited by any backend performance nor risk to overrun stuff.

Overkill IMO